homeassistant/components/backup/manager.py

```diff
@@ -778,6 +800,62 @@ def remove_subscription() -> None:
         return remove_subscription
 
 
+class KnownBackups:
+    """Track known backups."""
+
+    def __init__(self, manager: BackupManager) -> None:
+        """Initialize."""
+        self._backups: dict[str, KnownBackup] = {}
+        self._manager = manager
+
+    def load(self, stored_backups: dict[str, StoredKnownBackup]) -> None:
+        """Load backups."""
+        self._backups = {
+            backup_id: KnownBackup(
+                backup_id=backup_id, failed_agent_ids=backup["failed_agent_ids"]
+            )
+            for backup_id, backup in stored_backups.items()
+        }
+
+    def add(self, backup: AgentBackup, agent_errors: dict[str, Exception]) -> None:
```

Comment:
If we store the exception string we could show it to the user in the future.

Reply:
I'm not sure we should show the exception text. I think we could instead have some predefined exception types and store that?

Reply:
Ok, let's follow up with that when it's more clear.